server/services/yeelight/lib/device/discover.js

```diff
@@ -0,0 +1,68 @@
+const Promise = require('bluebird');
+const logger = require('../../../../utils/logger');
+const features = require('../features');
+const { DEVICE_EXTERNAL_ID_BASE } = require('../utils/constants');
+
+/**
+ * @description Send a broadcast to find the devices
+ * @returns {Promise<Array>} Resolve with array of new devices.
+ * @example
+ * discover();
+ */
+async function discover() {
+  return new Promise((resolve, reject) => {
+    if (!this.discoveryInProgress) {
+      logger.debug(`Yeelight: Discovery in progress...`);
+      const unknownDevices = [];
+      this.discoveryInProgress = true;
+
+      if (!this.discovery || this.discovery.isDestroyed) {
+        this.discovery = new this.yeelightApi.Discover();
+      }
+
+      // On device found...
+      this.discovery.on('deviceAdded', (discoveredDevice) => {
+        // ...check if it is already in Gladys...
+        const deviceInGladys = this.gladys.stateManager.get(
+          'deviceByExternalId',
+          `${DEVICE_EXTERNAL_ID_BASE}:${discoveredDevice.id}`,
+        );
+        if (deviceInGladys) {
+          logger.debug(`Yeelight: Device "${discoveredDevice.id}" is already in Gladys !`);
+        } else {
+          logger.debug(`Yeelight: Device "${discoveredDevice.id}" found, model: "${discoveredDevice.model}"`);
+          unknownDevices.push(features.getDevice(this.serviceId, discoveredDevice));
+        }
+      });
+
+      this.discovery.on('error', (error) => reject(error));
```

Comment:
I'm affraid here you add a listener each time you you run discover method.
Maybe you'll need to move all event subscription on a start method, and manager unsubscriptions on stop method.

Reply:
Only when the discovery process is running, but at the end of the discover method, I remove all subscriptions by calling `destroy()` method (see: [discover.ts](https://github.com/samuraitruong/yeelight/blob/master/src/discover.ts)).